src/main/java/reactor/netty/channel/MonoSendMany.java

```diff
@@ -391,10 +402,24 @@ void trySchedule(@Nullable Object data) {
 					}
 				}
 			}
+			else {
+				if (this.s == Operators.cancelledSubscription()) {
```

Comment:
because we will never go in the loop if loop has already exit seeing cancel. This is a racing case which causes memory leaks. The same problems were detected in FluxPublishOn, UnicastProcessor, FluxFlatMap, FLuxConcatMap and the only solution to ensure all the elements are discarded either - use SpMcQueue or do those checks

Reply:
of we will not check for cancel, then it will not be possible to guess at that point whether it is somebody doing work or it is 1 because it has already exited. In theory, I can use WIP field to do that check (e.g. if wip is negative == hurray, the stream is canceled - let's do a discard right immediately)

Reply:
if that question is optimizations matter - I can experiment with WIP as a cancel state, so we will get one less volatile read but may get few more volatile tries to writes in the drainLoop

Reply:
In theory that should be a good win.